controllers/envoyfleet_resources.go

```diff
@@ -0,0 +1,320 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gatewayv1alpha1 "github.com/kubeshop/kusk-gateway/api/v1alpha1"
+	"github.com/kubeshop/kusk-gateway/k8sutils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	envoyHTTPListenerPort  int32 = 8080
+	envoyHTTPSListenerPort int32 = 8443
+	envoyAdminListenerPort int32 = 19000
+)
+
+// EnvoyFleetResources is a collection of related Envoy Fleet K8s resources
+type EnvoyFleetResources struct {
+	client       client.Client
+	fleet        *gatewayv1alpha1.EnvoyFleet
+	configMap    *corev1.ConfigMap
+	deployment   *appsv1.Deployment
+	service      *corev1.Service
+	sharedLabels map[string]string
+}
+
+func NewEnvoyFleetResources(ctx context.Context, client client.Client, ef *gatewayv1alpha1.EnvoyFleet) (*EnvoyFleetResources, error) {
```

Comment:
Im not sure about this one

Its doing a lot. I feel it would be  better to have separate functions for creating each of the required resources. For transparency more than anything

Reply:
To specify dependencies, you specify parameters to the functions that depend on other things
```
configMap := createConfig()
deployment := createDeployment(configMap)
svc := createService(deployment)
```

For example ☝️

Reply:
or use Class :)

Reply:
@dobegor what do you think?

Reply:
I wish these functions would return objects that they generate and the final thing would be compose here rather than having everything done in a procedural style... However we can still pull this in and refactor it afterwards.